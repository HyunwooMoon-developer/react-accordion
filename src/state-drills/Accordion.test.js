import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordian from './Accordion';

describe('Accordian Component', ()=>{
    const sectionsProp = [
        {
            title: 'Section 1',
            content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
          },
          {
            title: 'Section 2',
            content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
          },
          {
            title: 'Section 3',
            content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
          }
    ];

    it('renders empty without error', () =>{
        const wrapper = shallow(<Accordian />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders empty given no sections', () =>{
        const wrapper = shallow(<Accordian sections={sectionsProp} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('open any clicked section', () =>{
        const wrapper = shallow(<Accordian sections={sectionsProp} />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('only opens one section at a time', () =>{
        const wrapper = shallow(<Accordian sections={sectionsProp} />)
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})