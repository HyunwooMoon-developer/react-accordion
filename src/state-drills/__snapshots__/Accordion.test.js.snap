// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component only opens one section at a time 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClickk={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClickk={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClickk={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component open any clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClickk={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClickk={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClickk={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component renders empty given no sections 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClickk={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClickk={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClickk={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component renders empty without error 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClickk={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClickk={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClickk={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
